### ISSUE TITLE
 locally ?

### ISSUE THREAD

**Issue by ghost** [thesephist/calamity#1]
`clear && python3.10 -m flask run`
```
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
 * Running on http://127.0.0.1:5000 (Press CTRL+C to quit)
127.0.0.1 - - [14/May/2022 03:46:29] "GET / HTTP/1.1" 404 -
```

 **Calamity** locally ? (wsgi ?)
`make serve` doesn't speak to model **(but is loaded)**, means no generation
```
(base) macos@iPro-von-MacOS calamity-main % make serve
oak src/main.oak
Calamity running at :7575
GET: /
GET: /css/main.css
GET: /js/torus.min.js
GET: /js/bundle.js
POST: /gen

```

**Comment by thesephist** [comment]
TLDR — `AI_INFERENCE_URL="http://localhost:5000" make serve`

---

So... this is where I have to apologize that the setup is kind of convoluted. As I note in the readme:

>There are two independent web apps: the model and API server, and the web app that wraps and calls the API and serves the client user interface. It's designed this way so that the API itself can be reused across other apps in my personal infrastructure, some of which aren't public yet.

But this does make it weird to run outside of my infra. To run it locally, you need to run both client and server apps locally and tell the client where the server is running.

When you start the client server (`make serve`) run it with the [env var telling it where the backend is running](https://github.com/thesephist/calamity/blob/main/calamity.service#L17) with `AI_INFERENCE_URL="http://localhost:5000"`. Since you're running it locally you don't need to mess with auth stuff, just set the URL. Then if you go to `localhost:7575`, you should see the "Generate" call make a `POST` to the Flask server, which means the two are talking to each other.

Hope that makes sense! Let me know if you run into more troubles.

**Comment by ghost** [comment]
I get response but no connection I get some **flask logs**
change of user wan't help `User=calamity-user` to `User=macos` 
maybe **oak** `ExecStart=/home/calamity-user/go/bin/oak ./src/main.oak` or
what about `AI_INFERENCE_TOKEN`
still testing `yes macos is a member of _www`

```
oak src/main.oak
Calamity running at :7575
GET: /
GET: /css/main.css
GET: /js/torus.min.js
GET: /js/bundle.js
POST: /gen
POST: /gen
POST: /gen
```

```
(AI) macos@iPro-von-MacOS service % python3.10 -m flask run
 * Environment: testing
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
 * Running on http://127.0.0.1:5000 (Press CTRL+C to quit)
127.0.0.1 - - [14/May/2022 20:23:34] "POST / HTTP/1.1" 404 -
127.0.0.1 - - [14/May/2022 20:25:13] "POST / HTTP/1.1" 404 -
127.0.0.1 - - [14/May/2022 20:25:17] "POST / HTTP/1.1" 404 -
```

**Comment by thesephist** [comment]
Ah, wait, sorry — I think I misspoke earlier because it had been a while since I set it up. `AI_INFERENCE_URL` needs to be the [full path to the API endpoint](https://github.com/thesephist/calamity/blob/main/src/main.oak#L24) which is `http://localhost:5000/generate/`.

**Comment by ghost** [comment]
`AI_INFERENCE_URL="http://localhost:5000/generate/" ./oak src/main.oak` 
changed to 
`AI_INFERENCE_URL="http://localhost:5000/generate" ./oak src/main.oak`
now get i green 
`127.0.0.1 - - [15/May/2022 17:59:19] "POST /generate HTTP/1.1" 308 -`

Model Kicks up and im waiting for first generation , but **seems that model not to full up.

![Bildschirmfoto 2022-05-15 um 17 56 30](https://user-images.githubusercontent.com/75278757/168482020-152ae0fa-d545-42ce-b418-e71b24633591.png)

but sad usage is maximum it is oak then the causer ? / model wrong loaded ?
![Bildschirmfoto 2022-05-15 um 20 55 55](https://user-images.githubusercontent.com/75278757/168489377-76339b98-9247-4d2e-862f-0dc6b019cec6.png)

**Comment by thesephist** [comment]
I'm not exactly sure what you mean by that last sentence, but if you're running the [full `gpt-j-6B` model](https://github.com/thesephist/calamity/blob/main/service/app.py#L5), it's quite a large model, and may take a long time for you to generate text on a slower computer. (So you may not see the Flask server respond for a long time). I run it on a server with 8 CPUs and 64GB of RAM, and it takes 15-30sec to generate 100 tokens/words. If your machine has less than 32GB of RAM, you won't be able to run that size model at all, and will need to use a smaller model from Huggingface Transformers.

**Comment by ghost** [comment]
i just mean that model runs, but strange maybe huggingface was do update of pytorch_model.bin,
and in [Calamity 🌋](https://github.com/thesephist/calamity) i see no outputs but error 
![Bildschirmfoto 2022-05-16 um 01 30 33](https://user-images.githubusercontent.com/75278757/168498961-1a51c058-4337-406d-a634-1231739dc114.png)

and I have no logs in `/var/log/calamity` with [permissions](https://github.com/thesephist/calamity/blob/a1906dd9a5f10f4f29825ec1e539139fcc03c01f/calamity.service#L27)

![Bildschirmfoto 2022-05-16 um 04 27 00](https://user-images.githubusercontent.com/75278757/168509680-f1c9e94a-f808-4a92-9144-42933d9b688b.png)

**Comment by thesephist** [comment]
I see that the Flask server is returning a 200 OK response to the `POST /generate/` request. That indicates to me that the model is generating responses correctly and Flask returns a correct response, and it is just not being returned to the browser. **If you open the Network pane in your browser's developer tools and see the request and response for the generate request, what do you see?**

If you're getting a "Load failed", it likely means the browser couldn't connect to the server (running on :7575). Maybe you're being timed out somewhere?

As a side note - if you're running things locally, nothing in `calamity.service`, including the log stuff, matters. That's only for deployment using systemd.

**Comment by ghost** [comment]
in Response is nothing 
but `oak build --entry src/app.js.oak --output static/js/bundle.js --web` without errors
chrome was jump automatic to section with `resp=fetch` line


![Bildschirmfoto 2022-05-16 um 19 40 11](https://user-images.githubusercontent.com/75278757/168651723-60a3c24a-cea4-4f00-b933-92ae9aa7072b.png)

**Comment by thesephist** [comment]
I see that in your screenshot, at the bottom there's an error that tells us the request to `/gen` is failing with the 403 Forbidden error code. This should mean that in the Networks panel in your DevTools ("Netzwerk" at the top of the screen in your screenshot) there should be something showing more data about that failed request.

I'm not sure where that "Forbidden" response is coming from, because I don't think any code in this repository ever generates a 403 response. To my understanding, if you're running these two commands at the same time:

- `flask run` inside `./service`
- `AI_INFERENCE_URL="http://localhost:5000/generate/" ./oak src/main.oak`

And open `localhost:7575`, everything _should_ just work. Your screenshot from [this comment](https://github.com/thesephist/calamity/issues/1#issuecomment-1127087360) makes me think that your backend (Flask) is receiving the right request, but your most recent screenshot tells me that it actually isn't receiving the request.... I'm not sure what to make of that discrepancy :/
